payu/models/cice.py

```diff
@@ -342,10 +342,29 @@ def archive(self, **kwargs):
         else:
             shutil.rmtree(self.work_input_path)
 
-        if self.expt.config.get('compress_logs', False):
+        if self.compression_enabled():
             self.compress_log_files()
 
+    def compression_enabled(self):
+        """
+        Determine whether to run log compression based on config.yaml settings.
+        Default to True when 'compress_logs' setting is absent.
+        """
+        archive_config = self.expt.config.get('archive', {})
+        if isinstance(archive_config, dict):
+            return archive_config.get('compress_logs', True)
+        else:
+            return True
+
```

Comment:
Similar to the `experiment.archiving()` method. This tries to keep backwards compatibility with `archive` being given as a boolean in the `config.yaml`. The cases are:

- No `archive` setting -> default to `compression_enabled = True`
- `archive` given as a boolean -> default to `compression_enabled = True`
- `archive` given as a dictionary but missing a `compress_logs` setting -> default to `compression_enabled = True`
- `archive` given as a dictionary and `compress_logs` setting included -> use specified value.

Does this behaviour seem reasonable?

Reply:
Yes. Very sensible approach.